app/enterprise/2.1.x/brain-immunity/model-training

```diff
@@ -0,0 +1,119 @@
+---
+title: Immunity Model Training
+---
+
+### Introduction
+
+Immunity automatically starts training its models once it is up and running and receiving data. Immunity will create a unique model for every unique endpoint + method combination it sees in incoming traffic. For example, if you have an endpoint [www.test-website.com/buy](http://www.test-website.com/buy) and traffic comes in with both GET and POST requests for that endpoint, Immunity will create two models; one for the endpoint + GET traffic and one for the endpoint + POST traffic.
+
+
+Our first model version gets created after the first hour and will continuously retrain itself for the first week to provide the best model possible. After that, every week all models retrain with a week of data.
+
+
+We also provide clients an endpoint to retrigger training themselves. We recommend retraining when the context of your app is expected to change a lot. For example, maybe there is an upcoming app release that will change several endpoints. If this is the case, one can POST to `http://<COLLECTOR_HOST>:<COLLECTOR_PORT>/resettrainer` to start the training cycle all over again.
+
+
+Let’s say you’d like slightly more control over the data your model sees, for example perhaps you know that weekend data is not particularly useful for model building because weekends are normally outliers that your team is prepared for. You can trigger model training for all models with a specified time period of data. Simply POST to `http://<BRAIN_HOST>:<COLLECTOR_PORT>/trainer`, with the start and end time of data you’d like included in training like this:
+
+```
+curl -d '{"start":"2019-01-08 10:00:00", "end":"2019-01-09 23:30:00"}' \
+ -H "Content-Type: application/json" \
+ -X POST http://<BRAIN_HOST>:<COLLECTOR_PORT>/trainer
+```
+or, in the browser like this:
+
+```
+http://<COLLECTOR_HOST>:<COLLECTOR_PORT>/trainer?start=&end=
+```
+** datetime value format: YYYY-MM-DD HH:mm:ss
+
+
+Additionally, you can specify the Kong service_id or route_id of the URLs you would like trained using the kong_entity parameter. Immunity would then only train URLs associated with the ID provided, and with the data specified by the start and end dates.
+
+```
+curl -d '{"start":"2019-01-08 10:00:00", "end":"2019-01-09 23:30:00", "kong_entity":"2beff163-061d-43ad-8d87-8f40d10805ba"}' \
+ -H "Content-Type: application/json" \
+ -X POST http://<COLLECTOR_HOST>:<COLLECTOR_PORT>/trainer
+```
+
+### Checking Models Trained
+
+Only endpoints + method combinations that have a model trained can be monitored for alerts. If you want to check which endpoints have models, you can use `http://<COLLECTOR_HOST>:<COLLECTOR_PORT>/monitoredendpoints` which return a list of all models in the system. Each item in this list contains the following identifying information for the model:
```

Comment:
```suggestion
Only endpoints and method combinations that have a model trained can be monitored for alerts. If you want to check which endpoints have models, you can use `http://<COLLECTOR_HOST>:<COLLECTOR_PORT>/monitoredendpoints`, which return a list of all models in the system. Each item in this list contains the following identifying information for the model:
```

use comma or switch "which" to "that" https://www.grammarly.com/blog/which-vs-that/